streaming/java/streaming-state/src/main/java/org/ray/streaming/state/strategy/AbstractStateStrategy.java

```diff
@@ -0,0 +1,80 @@
+package org.ray.streaming.state.strategy;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import org.ray.streaming.state.ITransactionState;
+import org.ray.streaming.state.StateException;
+import org.ray.streaming.state.StorageRecord;
+import org.ray.streaming.state.keystate.state.impl.AbstractState;
+import org.ray.streaming.state.serde.SerDeHelper;
+import org.ray.streaming.state.store.IKVStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class defines the StoreManager Abstract class.
+ * We use three layer to store the state, frontStore, middleStore and kvStore(remote).
+ */
+public abstract class AbstractStateStrategy<V> implements ITransactionState {
```

Comment:
According to the java doc, it looks like you want to use the strategy+proxy patterns to design the StoreManager. But the naming of  the interface and implementation classes are confused. What you tried to use is the strategy pattern, but what you really did is a template pattern. So here is the naming suggestion:

rename ITransactionState to TransactionStateStoreManager
rename AbstractStateStrategy<V> to AbstractTransactionStateStoreManager<V>
rename  AbstractTransactionState<V> to TransactionStateStoreManagerProxy<V>
rename CPStateStrategy<V> to CPTransactionStateStoreManager<V>
rename MVStateStrategy<V> to MVTransactionStateStoreManager<V>

thanks @TisonKun for asking about this.